sos/report/plugins/mellanox_firmware.py

```diff
@@ -0,0 +1,126 @@
+# Copyright (C) 2023 Nvidia Corporation, Alin Serdean <aserdean@nvidia.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, SoSPredicate, IndependentPlugin
+import os
+import time
+
+
+class MellanoxFirmware(Plugin, IndependentPlugin):
+
+    short_desc = 'Nvidia(Mellanox) firmware tools output'
+
+    plugin_name = "mellanox_firmware"
+    profiles = ('hardware', 'system')
+
+    def setup(self):
+        # Get all devices which have the vendor Mellanox Technologies
+        devices = []
+        device_list = self.collect_cmd_output('lspci -D -d 15b3::0200')
+        # Will return a string of the following format:
+        # 0000:08:00.0 Ethernet controller: Mellanox Technologies MT2892
+        # Family
+        if device_list['status'] != 0:
+            # bail out if there no Mellanox PCI devices
+            return
+
+        for line in device_list["output"].splitlines():
+            # Should return 0000:08:00.0
+            # from the following string
+            # 0000:08:00.0 Ethernet controller: Mellanox Technologies MT2892
+            # Family
+            devices.append(line[0:8]+'00.0')
+
+        devices = set(devices)
+
+        # Collect info from certain commands, only if mft package is present
+        # Mft package is present if OFED is installed
+        mft_pred = SoSPredicate(self, packages=['mft'])
```

Comment:
Setting the `packages` tuple to include `mft` will accomplish this a bit better.

Reply:
Unfortunately as stated above I can't limit just on `mft` or `mstflint` package.
I am unsure how can I achieve the above in a better way.

Reply:
The enablement triggers, like `packages`, operates on an `any` approach - so as long as at least one of the listed packages is installed, it will enable the plugin.